cas-server-support-osf/src/main/java/io/cos/cas/authentication/handler/support/OpenScienceFrameworkPrincipalFromRequestRemoteUserNonInteractiveCredentialsAction.java

```diff
@@ -434,20 +443,47 @@ protected OpenScienceFrameworkCredential constructCredential(
             }
 
             // AUTH TYPE 2.2: Institution login via the CAS protocol
-            // TODO: Add a check here to ensure that 1) `clientName` is not null and that 2) `clientName` matches one
-            //       of the configured institutions that uses the CAS protocol for delegation
             credential.setDelegationProtocol(DelegationProtocol.CAS_PAC4J);
             credential.setRemotePrincipal(Boolean.TRUE);
             credential.getDelegationAttributes().put("Cas-Identity-Provider", clientName);
             if (principal.getAttributes().size() > 0) {
                 for (final Map.Entry<String, Object> entry : principal.getAttributes().entrySet()) {
                     logger.debug(
-                            "[CAS PAC4J] User's institutional identity '{}' - auth header '{}': '{}'",
+                            "[CAS PAC4J] User's institutional identity '{}': '{}' with attribute '{}': '{}'",
+                            clientName,
                             principal.getId(),
                             entry.getKey(),
                             entry.getValue()
                     );
-                    credential.getDelegationAttributes().put(entry.getKey(), (String) entry.getValue());
```

Comment:
`(String) varName` casting only works if `varName` is of type `String` while `String.valueOf(varName())` cast ing works on both primitives and objects. We probably should only log and notify Sentry of multi-value attributes (e.g. `List`, `[]`) but still allow them and other attributes (e.g. `boolean`, `Integer`) that are not String to be put into the delegation map as `String`.